grpc/src/main/java/com/linecorp/armeria/common/grpc/GrpcStatusFunction.java

```diff
@@ -22,20 +22,18 @@
 
 import com.linecorp.armeria.common.annotation.UnstableApi;
 
-import io.grpc.Status;
-
 /**
- * A mapping function that converts a {@link Throwable} into a gRPC {@link Status}.
+ * A mapping function that converts a {@link Throwable} into an {@link ArmeriaGrpcStatus}.
  */
 @UnstableApi
 @FunctionalInterface
-public interface GrpcStatusFunction extends Function<Throwable, Status> {
+public interface GrpcStatusFunction extends Function<Throwable, ArmeriaGrpcStatus> {
 
     /**
-     * Maps the specified {@link Throwable} to a gRPC {@link Status}.
+     * Maps the specified {@link Throwable} to an {@link ArmeriaGrpcStatus}.
      * If {@code null} is returned, the built-in mapping rule is used by default.
      */
     @Nullable
     @Override
-    Status apply(Throwable throwable);
+    ArmeriaGrpcStatus apply(Throwable throwable);
```

Comment:
`ArmeriaGrpcStatus` is too Armeria specific type.
How about taking `Metadata` instead of introducing `ArmeriaGrpcStatus`?

```java
public interface GrpcStatusFunction extends BiFunction<Throwable, Metadata, Status> {

     Status apply(Throwable throwable, Metadata metadata);
}
```

A `Metadata` defined by users could be reflected by `metadata.merge(myMetadata)`.

```java
GrpcService.builder()
           .exceptionMapping((cause, metadata) -> {
               if (cause instanceof ...) {
                   metadata.merge(...)
                   return Status.XXX;
               }
           });
```

Reply:
Looks good.
Actually, introducing `ArmeriaGrpcStatus` looked a bit ugly to me.

Reply:
Removed `ArmeriaGrpcStatus`.